app/controllers/katello/concerns/api/v2/registration_controller_extensions.rb

```diff
@@ -24,6 +24,25 @@ def host_setup_extension
 
         super
       end
+
+      def context_urls
+        urls = super
+        urls[:rhsm_base_url] = rhsm_base_url
+        urls
+      end
+
+      private
+
+      def rhsm_base_url
+        smart_proxy = if params[:url]
+          SmartProxy.find_by(url: params[:url])
+        else
+          SmartProxy.pulp_primary
+        end
+
+        fail Foreman::Exception.new('Smart proxy content source not found!') unless smart_proxy
+        URI.join(smart_proxy.url, 'pulp/content')
```

Comment:
I'd expect RHSM on `/rhsm`

Reply:
When I install subscription manager with `katello_ca_consumer.rpm` the `/etc/rhsm/rhsm.conf` config file is set like this:
```
[rhsm]
# Content base URL:
baseurl = https://centos7-katello-devel-stable.example.com/pulp/content/
```

So I thought that it should be same

Reply:
That's the (Pulp) content URL, not RHSM itself. I'd consider the RHSM URL the basis for `subscription-manager config --server.hostname "$HOSTNAME" --server.port "$PORT" --server.prefix "$PREFIX"`.

Also, RHSM is always HTTPS while the Smart Proxy URL may be HTTP.

Reply:
Hmm, I'm getting super confused from all of the URLs

Right now there are:
* Foreman URL
* Smart Proxy URL
* Primary Pulp content URL
* Mirror pulp content URL
* RHSM URL

If I want to configure subscription-manager, what URLs should I use for these options:

```
subscription-manager config \
      --server.hostname="" \
      --server.port="" \
      --server.prefix="/rhsm" \
      --rhsm.baseurl=""
```

Reply:
It is my understanding (based on https://github.com/theforeman/puppet-foreman_proxy_content/blob/master/lib/puppet/provider/rhsm_reconfigure_script/rhsm_reconfigure_script.rb) that `--server.*` is the RHSM URL. `--rhsm.baseurl` is the Pulp content URL.

Reply:
Correct.

`server.hostname`: is the hostname subscription-manager should communicate with when making API calls
`server.port`: is the port subscription-manager should communicate with when making API calls
`server.prefix`: is a URL prefix that subscription-manager should talk to for all its API communication back to the server, in our case that is always `/rhsm`
`rhsm.baseurl`: is the location at which all content is located, this is prepended to all content paths when constructing the `redhat.repo` file

For example, if Candlepin knows a repo path is `/Default_Organization/Library/Mv_CV/repo_a/` and the `rhsm.baseurl` is `https://mycapsule.com/pulp/content` then the redhat.repo will get configured with:

```
baseurl=https://mycapsule.com/pulp/content/Default_Organization/Library/My_CV/repo_a
```